iceoryx_binding_c/test/moduletests/test_service_discovery.cpp

```diff
@@ -0,0 +1,174 @@
+{
+    ::testing::Test::RecordProperty("TEST_ID", "4bbd0b26-ed9d-4fcd-ae85-e7ea3783996d");
+    iox_pub_options_t options;
+    iox_pub_options_init(&options);
+    iox_pub_storage_t storage;
+    auto* publisher = iox_pub_init(&storage, "service", "instance", "event", &options);
+    ASSERT_NE(publisher, nullptr);
+    const iox_service_description_t SERVICE_DESCRIPTION = iox_pub_get_service_description(publisher);
+
+    const uint64_t SERVICE_CONTAINER_CAPACITY = 10U;
+    iox_service_description_t serviceContainer[SERVICE_CONTAINER_CAPACITY];
+    uint64_t missedServices = 0U;
+    const auto numberFoundServices = iox_service_discovery_find_service(sut,
+                                                                        SERVICE_DESCRIPTION.serviceString,
+                                                                        SERVICE_DESCRIPTION.instanceString,
+                                                                        SERVICE_DESCRIPTION.eventString,
+                                                                        serviceContainer,
+                                                                        SERVICE_CONTAINER_CAPACITY,
+                                                                        &missedServices,
+                                                                        MessagingPattern_PUB_SUB);
+
+    EXPECT_THAT(numberFoundServices, Eq(1U));
+    EXPECT_THAT(missedServices, Eq(0U));
+    EXPECT_THAT(serviceContainer[0U].serviceString, StrEq(SERVICE_DESCRIPTION.serviceString));
+    EXPECT_THAT(serviceContainer[0U].instanceString, StrEq(SERVICE_DESCRIPTION.instanceString));
+    EXPECT_THAT(serviceContainer[0U].eventString, StrEq(SERVICE_DESCRIPTION.eventString));
+}
+
+TEST_F(iox_service_discovery_test, FindServiceReturnsCorrectNumberOfServicesWhenServiceContainerTooSmall)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "01047b88-f257-447c-8c5e-9bef7c358433");
+    const uint64_t SERVICE_CONTAINER_CAPACITY = 3U;
+    iox_service_description_t serviceContainer[SERVICE_CONTAINER_CAPACITY];
+    uint64_t missedServices = 0U;
+    const auto numberFoundServices = iox_service_discovery_find_service(sut,
+                                                                        nullptr,
+                                                                        nullptr,
+                                                                        nullptr,
+                                                                        serviceContainer,
+                                                                        SERVICE_CONTAINER_CAPACITY,
+                                                                        &missedServices,
+                                                                        MessagingPattern_PUB_SUB);
+
+    EXPECT_THAT(numberFoundServices, Eq(SERVICE_CONTAINER_CAPACITY));
+    EXPECT_THAT(missedServices, Eq(6U - SERVICE_CONTAINER_CAPACITY));
```

Comment:
The 6 is actually `iox::NUMBER_OF_INTERNAL_PUBLISHERS` from the iceoryx_posh_types.hpp
(and strictly speaking the number we expect to find is the minimum of this number and the capacity of the output container, but nevermind.)